src/components/panel/panel.js

```diff
@@ -682,52 +709,80 @@ function MdPanelService($rootElement, $rootScope, $injector, $window) {
   /** @private @const */
   this._$window = $window;
 
+  /** @private {!Object<string, !MdPanelRef>} */
+  this._trackedPanels = {};
 
   /**
+   * @description
    * Default animations that can be used within the panel.
+   *
    * @type {enum}
    */
   this.animation = MdPanelAnimation.animation;
 
   /**
+   * @description
    * Possible values of xPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.xPosition = MdPanelPosition.xPosition;
 
   /**
+   * @description
    * Possible values of yPosition for positioning the panel relative to
    * another element.
+   *
    * @type {enum}
    */
   this.yPosition = MdPanelPosition.yPosition;
 }
 
 
 /**
+ * @description
  * Creates a panel with the specified options.
+ *
  * @param {!Object=} config Configuration object for the panel.
  * @returns {!MdPanelRef}
  */
 MdPanelService.prototype.create = function(config) {
   var configSettings = config || {};
 
+  if (angular.isDefined(configSettings.id) &&
```

Comment:
@EladBezalel You are correct about that. Is the `angular.isDefined()` method more expensive?